utils/src/field_helpers.rs

```diff
@@ -85,45 +110,97 @@ mod tests {
                 .is_ok(),
             true
         );
+
+        assert_eq!(
+            BaseField::to_hex(
+                BaseField::from_hex(
+                    "2eaedae42a7461d5952d27b97ecad068b698ebb94e8a0e4c45388bb613de7e08"
+                )
+                .expect("Failed to deserialize field hex")
+            ),
+            "2eaedae42a7461d5952d27b97ecad068b698ebb94e8a0e4c45388bb613de7e08"
+        );
     }
 
     #[test]
-    fn scalar_from_hex() {
+    fn field_bytes() {
         assert_eq!(
-            ScalarField::from_hex(""),
-            Err("Failed to deserialize field bytes")
+            BaseField::from_bytes(&[
+                46, 174, 218, 228, 42, 116, 97, 213, 149, 45, 39, 185, 126, 202, 208, 104, 182,
+                152, 235, 185, 78, 138, 14, 76, 69, 56, 139, 182, 19, 222, 126, 8
+            ])
+            .is_ok(),
+            true
         );
+
         assert_eq!(
-            ScalarField::from_hex(
-                "1428fadcf0c02396e620f14f176fddb5d769b7de2027469d027a80142ef8f07"
-            ),
-            Err("Failed to decode field hex")
+            BaseField::from_bytes(&[46, 174, 218, 228, 42, 116, 97, 213]),
+            Err("Failed to deserialize field bytes")
         );
+
         assert_eq!(
-            ScalarField::from_hex(
-                "0f5314f176fddb5d769b7de2027469d027ad428fadcf0c02396e6280142efb7d8"
+            BaseField::to_hex(
+                BaseField::from_bytes(&[
+                    46, 174, 218, 228, 42, 116, 97, 213, 149, 45, 39, 185, 126, 202, 208, 104, 182,
+                    152, 235, 185, 78, 138, 14, 76, 69, 56, 139, 182, 19, 222, 126, 8
+                ])
+                .expect("Failed to deserialize field bytes")
             ),
-            Err("Failed to decode field hex")
+            "2eaedae42a7461d5952d27b97ecad068b698ebb94e8a0e4c45388bb613de7e08"
         );
+    }
+
+    #[test]
+    fn field_bits() {
+        let fe =
+            BaseField::from_hex("2cc3342ad3cd516175b8f0d0189bc3bdcb7947a4cc96c7cfc8d5df10cc443832")
+                .expect("Failed to deserialize field hex");
+
+        let fe_check =
+            BaseField::from_bits(&fe.to_bits()).expect("Failed to deserialize field bits");
+        assert_eq!(fe, fe_check);
+
         assert_eq!(
-            ScalarField::from_hex(
-                "g64244176fddb5d769b7de2027469d027ad428fadcf0c02396e6280142efb7d8"
-            ),
-            Err("Failed to decode field hex")
+            BaseField::from_bits(&[
+                false, true, false, false, false, true, true, false, true, true, false, false,
```

Comment:
Can we convert the hex to bits in another fashion, to avoid hard-coding this?

Reply:
yes